From 72c9b841fccd70e9ece65295fb8945eece2e5d97 Mon Sep 17 00:00:00 2001
From: Patryk Obara <dreamer.tan@gmail.com>
Date: Sun, 14 Mar 2021 13:56:00 +0100
Subject: [PATCH] Use Meson summary to communicate configuration result

FluidSynth dependency requires Meson 0.53.0 to be on par with upstream
CMake buildsystem, so let's start summarizing build setup as well.
---
 meson.build         |  6 ++++--
 src/cpu/meson.build | 13 +++++++++----
 2 files changed, 13 insertions(+), 6 deletions(-)

diff --git a/meson.build b/meson.build
index 694a968c0..28b458bac 100644
--- a/meson.build
+++ b/meson.build
@@ -2,12 +2,14 @@ project('dosbox-staging', 'c', 'cpp',
         version : '0.77.0',
         license : 'GPL-2.0-or-later',
         default_options : ['cpp_std=c++14', 'b_ndebug=if-release'],
-        meson_version : '>= 0.51.0')
+        meson_version : '>= 0.53.0')
+
+summary('Build type',     get_option('buildtype'), section : 'Build Summary')
+summary('Install prefix', get_option('prefix'),    section : 'Build Summary')
 
 # After increasing the minimum-required meson version, make the following
 # improvements:
 #
-# - 0.53.0 - use summary() to communicate setup result
 # - 0.55.0 - subproject wraps are automatically promoted to fallbacks,
 #            stop using: "fallback : ['foo', 'foo_dep']" for dependencies
 # - 0.56.0 - use meson.current_source_dir() in unit tests
diff --git a/src/cpu/meson.build b/src/cpu/meson.build
index 848723b47..d8a74bc34 100644
--- a/src/cpu/meson.build
+++ b/src/cpu/meson.build
@@ -38,13 +38,18 @@ foreach line : core_selection
 endforeach
 
 if conf_data.has('C_DYNAMIC_X86')
-  message('Building dynamic core optimized for x86/x86_64 (dyn-x86) ' +
-          'for ' + conf_data.get('C_TARGETCPU'))
+  message('Building x86/x86_64-optimized dynamic core (dyn-x86) for ' +
+          conf_data.get('C_TARGETCPU'))
+  summary('Dynamic core', 'dyn-x86 (optimized for x86/x86_64)',
+          section : 'Emulator Features')
 elif conf_data.has('C_DYNREC')
-  message('Building dynamic core (dynrec) ' +
-          'for ' + conf_data.get('C_TARGETCPU'))
+  message('Building generic dynamic core (dynrec) for ' +
+          conf_data.get('C_TARGETCPU'))
+  summary('Dynamic core', 'dynrec (generic for ' + host_machine.cpu_family() + ')',
+          section : 'Emulator Features')
 else
   warning('Building without dynamic core support')
+  summary('Dynamic core', 'disabled', section : 'Emulator Features')
 endif
 
 message('Building for @0@-endian architecture'.format(host_machine.endian()))
